### PR TITLE
Remove empty ipv6 elements

### PR DESCRIPTION
**Pull request by ShimShtein**

### CODE REVIEW

**Comment by ShimShtein** [comment]
@adamruzicka, would be nice to test this one

File: lib/foreman_inventory_upload/generators/slice.rb [hunk 1]
```diff
@@ -105,8 +105,8 @@ def report_system_profile(host, host_ips_cache)
         @stream.array_field('network_interfaces') do
           @stream.raw(host.interfaces.map do |nic|
             {
-              'ipv4_addresses': [host_ips_cache[nic.ip]].compact,
-              'ipv6_addresses': [nic.ip6].compact,
+              'ipv4_addresses': [host_ips_cache[nic.ip]].reject(&:empty?),
+              'ipv6_addresses': [nic.ip6].reject(&:empty?),
```
Comment (adamruzicka) [inline lib/foreman_inventory_upload/generators/slice.rb:109]: How would I go about getting a nic where the address would be an empty string like you have in tests?

**Review by adamruzicka — COMMENTED** [review]
In general I'm in favor, I'm just curious how we can get into a situation where this is needed.

File: lib/foreman_inventory_upload/generators/slice.rb (hunk 1 above)
Comment (ShimShtein) [reply lib/foreman_inventory_upload/generators/slice.rb:109]: I suppose it comes from hosts that are created in foreman from imported facts. We have a customer that we have seen this behaviour. I can get more details, if you like.

Comment (adamruzicka) [reply lib/foreman_inventory_upload/generators/slice.rb:109]: Please do

**Review by adamruzicka — APPROVED** [review]
ACK

**Comment by ShimShtein** [comment]
@laviro, ready for review/merge

**Review by laviro — APPROVED** [review]

**Comment by laviro** [comment]
Thanks @ShimShtein !

**Pull request merged**